device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/IotHubReceiveTask.java

```diff
@@ -3,36 +3,48 @@
 
 package com.microsoft.azure.sdk.iot.device.transport;
 
+import com.microsoft.azure.sdk.iot.device.CustomLogger;
+
 /**
  * Polls an IoT Hub for messages and invokes a callback if one is found.
  * Meant to be used with an executor that continuously calls run().
  */
 public final class IotHubReceiveTask implements Runnable
 {
     protected final IotHubTransport transport;
-
+    /**
+     * Private logger for class
+     */
+    private final CustomLogger logger = new CustomLogger(this.getClass());
+    		
     public IotHubReceiveTask(IotHubTransport transport)
     {
         // Codes_SRS_IOTHUBRECEIVETASK_11_001: [The constructor shall save the transport.]
         this.transport = transport;
+        
+        if (transport == null)
+        	throw new IllegalArgumentException("Parameter 'transport' must not be null");
+        
```

Comment:
@prmathur-microsoft : With current setup constructors of IotHubSendTask and IotHubReceiveTask will write a wrong log error message when parameter 'transport' is not null. If it's null, no log message will be written, since the code is no longer reachable